meshtastic/serial_interface.py

```
@@ -36,8 +36,24 @@ class SerialInterface(StreamInterface):
+
+        Raises
+        ------
+        MeshInterfaceError
+            If an explicit path is empty/whitespace-only or if multiple ports are
+            detected without an explicit ``--port`` selection.
+        """
         if self.devPath is not None:
+            if not self.devPath.strip():
+                raise self.MeshInterfaceError(
+                    "Serial port path cannot be empty; pass None to auto-detect."
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)